samples/nextjs/src/pages/[[...path]].tsx

```diff
@@ -30,24 +31,41 @@ const SitecorePage = ({
     return <NotFound />;
   }
 
-  useEffect(() => {
-    // Do not trigger client tracking when pages are requested by Sitecore XP instance:
-    // - no need to track in Edit and Preview modes
-    // - in Explore mode all requests will be tracked by Sitecore XP out of the box
-    if (isPreview) return;
-
-    trackingService
-      .trackCurrentPage(layoutData.sitecore.context, layoutData.sitecore.route)
-      // eslint-disable-next-line @typescript-eslint/no-explicit-any
-      .catch((error: any) => console.error('Tracking failed: ' + error.message));
-  }, [isPreview, layoutData]);
-
   const context: StyleguideSitecoreContextValue = {
     route: layoutData.sitecore.route,
     itemId: layoutData.sitecore.route?.itemId,
     ...layoutData.sitecore.context,
   };
 
+  // Start loading personalization before render occurs, personalization loading components rely in service state
+  // Do not load personalization twice for pages with query, see Caveats for dynamic routes in Next.js doc
+  useMemo(() => {
+    const disconnectedMode =
+      layoutData.sitecore.route &&
+      layoutData.sitecore.route.layoutId === 'available-in-connected-mode';
+    if (disconnectedMode) {
+      return;
+    }
+    // Load personalization client side only
+    if (!process.browser) {
+      return;
+    }
+    // Do not trigger client tracking when pages are requested by Sitecore XP instance:
+    // - no need to track in Edit and Preview modes
+    // - in Explore mode all requests will be tracked by Sitecore XP out of the box
+    if (isPreview) {
+      return;
+    }
+
+    layoutPersonalizationService.loadPersonalization(context, context.route).then((p) => {
```

Comment:
How a front-end developer can pass the result of self-implemented tracking consent?
Should loadPersonalization accept the third param: track (true|false)?

Reply:
In next PBI we are going to extend loadPersonalization with requestOptions where customer is able to pass tracking=true/false